editor-components/Image_gallery/image_gallery_5/image-gallery5.tsx

```diff
@@ -0,0 +1,131 @@
+import * as React from "react";
+import { BaseImageGallery } from "../../EditorComponent";
+import styles from "./image-gallery5.module.scss";
+
+interface ImageType {
+  type: string;
+  key: string;
+  displayer: string;
+  value: string;
+}
+
+class ImageGalleryComponent5 extends BaseImageGallery {
+  constructor(probs?: any) {
+    super(probs, styles);
+
+    this.addProp({
+      type: "array",
+      key: "gallery",
+      displayer: "Gallery",
+      value: [
+        {
+          type: "image",
+          key: "imageGallery",
+          displayer: "Image1",
+          value: "https://arty.flatheme.net/assets/images/p-clean-2-1.jpg"
+        },
+        {
+          type: "image",
+          key: "imageGallery",
+          displayer: "Image2",
+          value: "https://arty.flatheme.net/assets/images/p-clean-2-2.jpg"
+        },
+        {
+          type: "image",
+          key: "imageGallery",
+          displayer: "Image3",
+          value: "https://arty.flatheme.net/assets/images/p-clean-2-3.jpg"
+        },
+        {
+          type: "image",
+          key: "imageGallery",
+          displayer: "Image4",
+          value: "https://arty.flatheme.net/assets/images/p-clean-2-4.jpg"
+        },
+        {
+          type: "image",
+          key: "imageGallery",
+          displayer: "Image5",
+          value: "https://arty.flatheme.net/assets/images/p-clean-2-5.jpg"
+        },
+        {
+          type: "image",
+          key: "imageGallery",
+          displayer: "Image6",
+          value: "https://arty.flatheme.net/assets/images/p-clean-2-6.jpg"
+        }
+      ]
+    });
+
+    this.setComponentState("is_image_clicked", false);
+    this.setComponentState("clicked_image_index", 0);
+  }
+
+  getName(): string {
+    return "Image Gallery 5";
+  }
+
+  handleImageClick(index: number) {
+    this.setComponentState("is_image_clicked", true);
+    this.setComponentState("clicked_image_index", index);
+  }
+
+  handleCloseClick() {
+    this.setComponentState("is_image_clicked", false);
+  }
+
+  handleNextImage() {
+    const galleries = this.getPropValue("gallery");
+    let currentIndex = this.getComponentState("clicked_image_index");
+    currentIndex = (currentIndex + 1) % galleries.length;
+    this.setComponentState("clicked_image_index", currentIndex);
+  }
+
+  handlePrevImage() {
+    const galleries = this.getPropValue("gallery");
+    let currentIndex = this.getComponentState("clicked_image_index");
+    currentIndex = (currentIndex - 1 + galleries.length) % galleries.length;
+    this.setComponentState("clicked_image_index", currentIndex);
+  }
+
+  render() {
+    const galleries = this.getPropValue("gallery");
```

Comment:
Change with the type that you have written.
```javascript
const galleries = this.castToObject<ImageType[]>("gallery");
```

[Please read.](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md)